packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -806,6 +829,26 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     if (_snackBars.isNotEmpty)
       _addIfNonNull(children, _snackBars.first._widget, _ScaffoldSlot.snackBar);
 
+    if (config.persistentFooterButtons != null) {
```

Comment:
Yeah, I figured it would be better to treat an empty list by drawing the bar and leaving a null list as the way to get rid of the bar.